akka-actor/src/main/scala/akka/io/TcpConnection.scala

```diff
@@ -335,6 +335,10 @@ private[io] abstract class TcpConnection(val tcp: TcpExt, val channel: SocketCha
         if (TraceLogging) log.debug("setSoLinger(true, 0) failed with [{}]", e)
     }
     channel.close()
+    // On linux, closing sends the RST directly. On windows, however, the connection is
+    // closed only when the channel is deregistered from the selector, which happens before
+    // and after 'select' polling.
```

Comment:
Where does it happen? We don't ever explicitly call `key.cancel()`. I think what we otherwise rely on is that the channel calls `cancel` itself internally on connection closure. Did you find any bug tickets that describe this behavior?